### ISSUE TITLE
DockerClient.images.list(**kwargs) ignores **kwargs on docker 20

### ISSUE THREAD

**Issue by eliorc** [docker/docker-py#2752]
# Bug Description

Using the docker client, we can list the existing images using [`.image.list`.](https://docker-py.readthedocs.io/en/stable/images.html#docker.models.images.ImageCollection.list)
In the documentation it reads that using the `name` argument, we can filter the list to the name specified so for example 

This 

```python3
import docker

client = docker.from_env()

print(client.images.list())
```

Should output a different output than this

```python3
import docker

client = docker.from_env()

print(client.images.list(name='non_existent/docker:image'))
```

But the output is the same for both code snippets, on version docker 20.10.2 (docker-py version 4.4.0)

# Bug setup and reproduction

This bug does not exist in version 19 (can't say exatly which verison), but using version 20.10.2, build 2291f61 and docker-py 4.4.0 this bug appears.
To replicate it, assuming you have some images on your machine run the following

```python3
import docker

client = docker.from_env()

assert [] == client.images.list(name='non_existent/docker:image')
```

Thanks to @michalbarer for uncovering this bug

**Comment by feliperuhland** [comment]
Hi @eliorc 

I could not reproduce the issue.
I'm running docker engine API v1.41.

Could you send me your docker API version? You can get it running `docker version`.

From the last release notes, I could notice this change that could be related:

> The filter (singular) query parameter, which was deprecated in favor of the filters option in Docker 1.13, has now been removed from the GET /images/json endpoint. The parameter remains available when using API version 1.40 or below.

**Comment by feliperuhland** [comment]
I believe that issue was resolved here: https://github.com/docker/docker-py/commit/1757c974fa3a05b0e9b783af85242b18df09d05d

**Comment by Nicceboy** [comment]
I had similar issue, and it works now after upgrading to latest release.

**Comment by eliorc** [comment]
> Could you send me your docker API version? You can get it running `docker version`.

@feliperuhland I supplied all the versions in my original post (docker version being 20.10.2)

I can too confirm that this bug is resolved on my current `docker-py` version (4.4.4)
